pkgs/unified_analytics/lib/src/session.dart

```diff
@@ -19,19 +17,19 @@ class Session {
   final File sessionFile;
   final ErrorHandler _errorHandler;
 
-  late int _sessionId;
-  late int _lastPing;
+  int _sessionId;
 
   Session({
     required this.homeDirectory,
     required this.fs,
     required ErrorHandler errorHandler,
   })  : sessionFile = fs.file(p.join(
             homeDirectory.path, kDartToolDirectoryName, kSessionFileName)),
+        _sessionId = DateTime.now().millisecondsSinceEpoch,
```

Comment:
this seems like it's "solving" the problem of LateInitializationError by assigning the variable the wrong value. Or am I misunderstanding how this works?

Reply:
You're correct, this value is getting set when calling the constructor and it will eventually get replaced when we run `Session.initialize`. This just makes it so that we don't have to make the variable `late`.

Reply:
> we could be setting the this value in the file because we need a new session id if the old one had 30 mins of inactivity

But if it has NOT been 30 minutes of inactivity, the correct session ID would be a different value, right?

Reply:
Yes that's correct, we would have to decide if that is worth it to remove the `late` variable

Reply:
sgtm, no rush

Reply:
from discussion, maybe we should just make _sessionId nullable, and allow sending a null session id. this would also help when sending errors.

Reply:
Yep this makes sense with me, i also just confirmed that sending `null` for the session id is a valid json payload for GA4